assets/dev/js/admin/floating-elements/new-floating-elements.js

```diff
@@ -0,0 +1,66 @@
+import NewFloatingelementsLayout from 'elementor-admin/floating-elements/layout';
+
+const NewFloatingElementsModule = elementorModules.ViewModule.extend( {
+
+	getDefaultSettings() {
+		return {
+			selectors: {
+				addButtonTopBar: '.page-title-action',
+				addButtonAdminBar: '#wp-admin-bar-new-e-floating-buttons a',
+				addButtonEmptyTemplate: '#elementor-template-library-add-new',
+			},
+		};
+	},
+
+	getDefaultElements() {
+		var selectors = this.getSettings( 'selectors' );
+
+		return {
+			addButtonTopBar: document.querySelector( selectors.addButtonTopBar ),
+			addButtonAdminBar: document.querySelector( selectors.addButtonAdminBar ),
+			addButtonEmptyTemplate: document.querySelector( selectors.addButtonEmptyTemplate ),
+		};
+	},
+
+	bindEvents() {
+		if ( this.elements.addButtonTopBar !== null ) {
+			this.elements.addButtonTopBar.addEventListener( 'click', this.onAddButtonClick );
+		}
+
+		if ( this.elements.addButtonAdminBar !== null ) {
+			this.elements.addButtonAdminBar.addEventListener( 'click', this.onAddButtonClick );
+		}
+
+		if ( this.elements.addButtonEmptyTemplate !== null ) {
+			this.elements.addButtonEmptyTemplate.addEventListener( 'click', this.onAddButtonClick );
+		}
+
+		elementorCommon.elements.$window.on( 'hashchange', this.showModalByHash.bind( this ) );
+	},
+
+	showModalByHash() {
+		if ( '#add_new' === location.hash ) {
+			this.layout.showModal();
+
+			location.hash = '';
+		}
+	},
+
+	onInit() {
+		elementorModules.ViewModule.prototype.onInit.apply( this, arguments );
```

Comment:
`super`?

Reply:
@louiswol94 I tried refactoring this to use a class but this doesn't work as the object at the top of the hierarchy is a function, not a class. so extending this with a class and using super doesn't work